pkg/queue/cluster_queue_best_effort_fifo.go

```diff
@@ -69,6 +69,16 @@ func (cq *ClusterQueueBestEffortFIFO) Delete(w *kueue.Workload) {
 	cq.ClusterQueueImpl.Delete(w)
 }
 
+func (cq *ClusterQueueBestEffortFIFO) DeleteFromQueue(q *Queue) {
+	for _, w := range q.items {
+		key := workload.Key(w.Obj)
+		if wl := cq.inadmissibleWorkloads[key]; wl != nil {
+			delete(cq.inadmissibleWorkloads, key)
+		}
+		cq.ClusterQueueImpl.Delete(w.Obj)
```

Comment:
nit: Instead, do `cq.ClusterQueueImpl.DeleteFromQueue(q)` at the end of the loop.

That should be a bit more future-proof, in case the implementation of ClusterQueueImpl changes.